pyccel/parser/semantic.py

```diff
@@ -1108,11 +1148,36 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign,ar
             if not arr_in_multirets:
                 self._ensure_target(rhs, d_lhs)
 
-            var = self.check_for_variable(name)
-
+            if isinstance(lhs, DottedName):
+                var = self.check_for_attribute(lhs)
+            else:
+                var = self.check_for_variable(name)
             # Variable not yet declared (hence array not yet allocated)
             if var is None:
-
+                if isinstance(lhs, DottedName):
+                    if (self._current_function == '__init__' or self.current_class) and lhs.name[0] == 'self':
+                        cls      = self.get_variable('self')
```

Comment:
```suggestion
                        var      = self.get_variable('self')
```
This is recollected below

Reply:
This is still collected twice